### ISSUE TITLE
For the fine-tuning in NLG task, why not use standard language model objective with teacher forcing?

### ISSUE THREAD

**Issue by yuewang-cuhk** [microsoft/unilm#21]
Hi, I would like to know more about the comparison between the standard language model fine-tuning (teacher forcing) and the masked language model fine-tuning (same objective as pre-training) in your paper. In my opinion, for the text generation task, the most popular fine-tuning approach would be the teacher forcing in language modeling as it mimics the generation process during testing. Thanks!

**Comment by donglixp** [comment]
Hi @yuewang-cuhk ,

The seq2seq fine-tuning process is also teacher forcing. The model learns to predict a token by conditioning on the previously generated tokens and all the input tokens.

**Comment by yuewang-cuhk** [comment]
Oh, I see. But in your setting, you only predict for the masked tokens. Why not predict for all the tokens like the traditional seq2seq training? It seems that the latter can make full use of the training resources.

**Comment by yuewang-cuhk** [comment]
@f-lng  Thanks for the explanation! I also agree with you. It would be much more efficient by keeping the same fine-tuning objective as the pre-training. If not, it seems that we need to replicate one instance with an output sequence length of N into N training instances.
`E.g., (a1 a2 a3 b1 b2 b3)->(a1, a2, a3, [mask]), (a1, a2, a3, b1, [mask]), (a1, a2, a3, b1, b2, [mask])`
@donglixp Are we correct? I am also curious about whether there are experiments or mathematical proof showing that two different aforementioned fine-tuning settings can achieve comparable performance? Thanks!

**Comment by f-lng** [comment]
@yuewang-cuhk
Have a look at the paper "Exploring the Limits of Transfer Learning with a Unified Text-to-Text Transformer". It seems that an Encoder-Decoder Architecture gives the best results for a given computational budget (tough, it has double the parameters).
